### ISSUE TITLE
Why does `topbar` eat click events on touch devices?

### ISSUE THREAD

**Issue by workmanw** [zurb/foundation#6925]
See: https://github.com/zurb/foundation/blob/29a7102bc1a0e07d5792d9c5753112e1a4cc3504/js/foundation/foundation.topbar.js#L196

This is causing an issue for our users with Windows Laptops that touch capable. We install event handlers on the items in our dropdown list and those event handlers aren't firing for this reason.

I'm not suggesting this is a bug, just seeking clarification on the reasoning for this. Thanks!

**Comment by rafibomb** [comment]
Looks like this is an issue with how Windows handles touch devices - that is to say it does not. Modernizr is being used to help detect touch devices but Windows doesn't work there. 

You can find more info on a previous thread https://github.com/zurb/foundation/issues/4346

**Comment by workmanw** [comment]
@rafibomb That doesn't answer the question though. My question wasn't why is `Modernizr.touch` true. It was why are click events swallowed for touch devices. We have the same problem on iPads. User's can't use our topbar menu due to our event handlers aren't firing because Foundation is swallowing them.

**Comment by workmanw** [comment]
@rafibomb I mean no disrespect here. 

I didn't just open this issue arbitrarily. I am trying to use your toolkit and I'm trying to understand the reason for a certain behavior that's causing undesirable behavior in our app. 

Before opening this issue, I Google'd for an explanation, I read the docs, I looked for code comments and I traced down the original commit that added this code [3659881e](https://github.com/zurb/foundation/commit/3659881e75afc990881535870173e6e9ceb3137c#diff-258c5a99cb3cd5afe2b68c1f53a8a298). All of which yield no answer.

I still have no answer and my issue was closed. This does not inspire confidence.
